File: C/牛客网/剑指offer/item045.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
	bool IsContinuous(vector<int> numbers) {
		if (numbers.size() != 5) return false;
		sort(numbers.begin(), numbers.end());
		int count = 0;
		for (int i = 0; i < numbers.size(); i++)
			if (numbers[i] == 0)
				count++;
		for (int i = count; i < numbers.size() - 1; i++)
			if (numbers[i] == numbers[i + 1])
				return false;
		return numbers[numbers.size() - 1] - numbers[count] < numbers.size();		
	}
};

File: Java/LeetCode/100/050/item045.java
public class Solution {
    public int jump(int[] nums) {
        int count = 0, end = 0, fathest = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if (fathest < nums[i] + i) fathest = nums[i] + i;
            if (i == end) {
                count++;
                end = fathest;
            }
        }
        return count;
    }
}